app/src/app/[lng]/auth/signup/page.tsx

```diff
@@ -42,11 +43,20 @@ export default function Signup({
     handleSubmit,
     register,
     setError: setFormError,
+    setValue,
     formState: { errors, isSubmitting },
   } = useForm<Inputs>();
 
   const [error, setError] = useState("");
 
+  const searchParams = useSearchParams();
+  const params = searchParams.get("callbackUrl");
```

Comment:
Rename this to `callbackUrl` so it's easier to understand. You have an URL string here, not a list of parameters.